src/chapters/05/main.tex

```diff
@@ -0,0 +1,279 @@
+
+\begin{pyin}
+import sympy as sym
+
+t = sym.Symbol("t")
+alpha = sym.Symbol("alpha")
+I_0 = sym.Symbol("I_0")
+I = sym.Function("I")
+
+
+def get_equation(alpha=alpha):
+    """Return the differential equation.
+
+    Args:
+        alpha: a float (default: symbolic alpha)
+
+    Returns:
+        A symbolic equation
+    """
+    return sym.Eq(sym.Derivative(I(t), t), -alpha * I(t))
+\end{pyin}
+
+Using this we can get the equation that defines the population change over time:
+
+\begin{pyin}
+eq = get_equation()
+print(eq)
+\end{pyin}
+
+which gives:
+
+\begin{pyout}
+Eq(Derivative(I(t), t), -alpha*I(t))
+\end{pyout}
+
+Note that if you are using Jupyter then your output will actually be a
+well rendered mathematical equation:
+
+\[
+\frac{d}{d t} I{\left(t \right)} = - \alpha I{\left(t \right)}
+\]
+
+Note that we can pass a value to \(\alpha\) if we want to:
+
+\begin{pyin}
+eq = get_equation(alpha=1)
+print(eq)
+\end{pyin}
+
+\begin{pyout}
+Eq(Derivative(I(t), t), -I(t))
+\end{pyout}
+
+We will now write a function to obtain the solution to this differential
+
+\begin{pyin}
+def get_solution(I_0=I_0, alpha=alpha):
+    """Return the solution to the differential equation.
+
+    Args:
+        I_0: a float (default: symbolic I_0)
+        alpha: a float (default: symbolic alpha)
+
+    Returns:
+        A symbolic equation
+    """
+    eq = get_equation(alpha=alpha)
+    return sym.dsolve(eq, I(t), ics={I(0): I_0})
+\end{pyin}
+
+We can verify the solution discussed previously:
+
+\begin{pyin}
+sol = get_solution()
+print(sol)
+\end{pyin}
+
+which gives:
+
+\begin{pyout}
+Eq(I(t), I_0*exp(-alpha*t))
+\end{pyout}
+
+\[I(t) = I_0 e ^{-\alpha t}\]
+
+We can use sympy itself to verify the result, by taking the derivative of the
+right hand side of our solution.
+
+\begin{pyin}
+print(sym.diff(sol.rhs, t) == -alpha * sol.rhs)
+\end{pyin}
+
+which gives:
+
+\begin{pyout}
+True
+\end{pyout}
+
+All of the above has given us the general solution in terms of \(I(0)=I_0\) and
+\(\alpha\) however we have written the code in such a way as we can pass the
+actual parameters:
+
+\begin{pyin}
+sol = get_solution(alpha=2, I_0=100)
+print(sol)
+\end{pyin}
+
+which gives:
+
+\begin{pyout}
+Eq(I(t), 100*exp(-2*t))
+\end{pyout}
+
+Now, to calculate the cost we will write a function to integrate our result:
+
+\begin{pyin}
+def get_cost_without_cure(
+    I_0=I_0, alpha=alpha, cost_per_person=10
+):
+    """Return the cost.
+
+    Args:
+        I_0: a float (default: symbolic I_0)
+        alpha: a float (default: symbolic alpha)
+        cost_per_person: a float (default: 10)
+
+    Returns:
+        A symbolic expression
+    """
+    I_sol = get_solution(I_0=I_0, alpha=alpha)
+    return (
+        sym.integrate(I_sol.rhs, (t, 0, sym.oo))
+        * cost_per_person
+    )
+\end{pyin}
+
+The cost with cure can use the above but needs to modify the rate \(\alpha\) and
+include the one off cost per person:
+
+\begin{pyin}
+def get_cost_with_cure(
```

Comment:
Remove this and use just one cost function with cost of cure set to zero as default.